packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -221,9 +222,12 @@ class IosProject extends XcodeBasedProject {
   /// The return future will resolve to string path to the output file.
   Future<String> outputsUniversalLinkSettings({
     required String configuration,
-    required String scheme,
-    required String target,
+    String? scheme,
```

Comment:
Actually I am not sure if we should support scheme at all. My understanding of scheme is that it contains a set of targets. I am not sure what does it mean to get build settings for a scheme.

Reply:
@andrewkolos can you review this PR? I'm still fuzzy on the relationship between targets and schemes.

Reply:
Hah, I hardly know the difference myself, but—no worries—I'll take a look.

Reply:
As I understand, projects and targets can have _Build Settings_, while schemes can have a _Build Configuration_. Settings are applied in this order: project settings, build configuration settings, then target settings[^1].

[^1]: https://developer.apple.com/documentation/xcode/adding-a-build-configuration-file-to-your-project#Map-build-settings-to-a-build-configuration

Reply:
> `the show build settings xcode command can only accept one of the target or scheme flag. Therefore I make them optional.`

Running `xcodebuild -scheme iosapp -configuration jk -target iosapp -showBuildSettings` in one of my local projects doesn't result in any error. Are you sure that the command only accepts exactly one of `-scheme` and `-target`?

Reply:
So `getBuildSettings` specifies `-project`, which if `-project` is specified, you can only use either `-scheme` or `-target`. However, if you were to exclude the `-project` flag and use `-scheme` and `-target` together, it basically just ignores the `-target` flag.

So @andrewkolos `xcodebuild -scheme iosapp -configuration jk -target iosapp -showBuildSettings` works because `-project` is not specified.

Also, Build Configuration is not per scheme but per project.

When `-scheme` is specified, it shows the build settings for all runnable targets. This can be seen if you edit the scheme and check the `Run` column for more than one targets.

![Screenshot 2023-09-13 at 10 44 01 AM](https://github.com/flutter/flutter/assets/15619084/00971e45-e547-44f3-ad00-03326b70a0af)

So if I ran `xcodebuild -project xxx -configuration xxx -scheme Runner -showBuildSettings`, it would show something like:

```
Build settings for action build and target Runner:
...
Build settings for action build and target RunnerTests:
...
```

When `-target` is set, it will only show the build settings for only the set target.

Reply:
Looking at our code that parses the build settings, looks like we don't really handle multiple targets... looks like it operates under the assumption that a scheme will only have a single target.

I think there are definitely improvements that could be made here, but probably out of scope for this PR. I filed an issue to follow up on this https://github.com/flutter/flutter/issues/134669.

For this PR, I think it would make sense to only support for target

Reply:
Looking at our code, seems like we don't really handle multiple targets per scheme. It operates under the assumption that there's only one target per scheme.

I think there are improvements that could be made here, but probably out of scope for this PR. I filed an issue to follow up on that: https://github.com/flutter/flutter/issues/134669

For your PR, since `-configuration` is required, I think it's okay to only use `-target`. 

If `-configuration` was not set, you would have needed `-scheme` to get the correct `-configuration`.

Reply:
Sounds good, thanks for explanation. I will update the code.